### PR TITLE
cmd/scollector: fix multiple service tags for blade servers in vsphere

### PR DESCRIPTION
**Pull request by gbrayut**
Fixes issue with multiple service tags on new VM servers (see ny-bosun01 logs)

### CODE REVIEW

File: cmd/scollector/collectors/vsphere.go [hunk 1]
```diff
@@ -181,9 +183,10 @@ func vsphereHost(v *vsphere.Vsphere, md *opentsdb.MultiDataPoint) error {
 							return err
 						}
 						if t.IdentiferType.Key == "ServiceTag" {
-							metadata.AddMeta("", tags, "serialNumber", t.IdentiferValue, false)
+							lastServiceTag = t.IdentiferValue
 						}
 					}
+					metadata.AddMeta("", tags, "serialNumber", lastServiceTag, false)
```
Comment (mjibson) [inline cmd/scollector/collectors/vsphere.go]: lastServiceTag could be empty if the for loop never ran or something. Instead, we should just have  `serviceTagDone := false` before the for loop, then if the bool is false, set the metadata where it is now, then set the bool to true.

File: cmd/scollector/collectors/vsphere.go [hunk 2]
```diff
@@ -171,6 +171,8 @@ func vsphereHost(v *vsphere.Vsphere, md *opentsdb.MultiDataPoint) error {
 				switch p.Name {
 				case "summary.hardware.otherIdentifyingInfo":
 					d := xml.NewDecoder(bytes.NewBufferString(p.Val.Inner))
+					//Blade servers may have multiple service tags. Use the last one.
```
Comment (mjibson) [inline cmd/scollector/collectors/vsphere.go]: space after //

Comment (mjibson) [inline cmd/scollector/collectors/vsphere.go]: and if we make that change below then this should say the first instead of last

Comment (gbrayut) [inline cmd/scollector/collectors/vsphere.go]: We want the last one... first one is for the blade enclosure.

Comment (mjibson) [inline cmd/scollector/collectors/vsphere.go]: In that case, only add it if lastServiceTag != ""

**Pull request merged**